### PR TITLE
[azure] New azure pack

### PR DESCRIPTION
**Pull request by Kami**
It comes with an initial set of compute and storage actions.

```bash
+----------------------+-------+----------------------+----------------------+
| ref                  | pack  | name                 | description          |
+----------------------+-------+----------------------+----------------------+
| azure.create_contain | azure | create_container     | Create a new storage |
| er                   |       |                      | container.           |
| azure.create_vm      | azure | create_vm            | Create a new VM.     |
| azure.delete_contain | azure | delete_container     | Delete a storage     |
| er                   |       |                      | container.           |
| azure.delete_object  | azure | delete_object        | Delete an object.    |
| azure.destroy_vm     | azure | destroy_vm           | Destroy a VM.        |
| azure.list_container | azure | list_container_objec | List storage objects |
| _objects             |       | ts                   | for the provided     |
|                      |       |                      | container.           |
| azure.list_container | azure | list_containers      | List storage         |
| s                    |       |                      | containers.          |
| azure.list_vms       | azure | list_vms             | List available VMs.  |
| azure.reboot_vm      | azure | reboot_vm            | Reboot a running VM. |
| azure.upload_file    | azure | upload_file          | Upload a file to the |
|                      |       |                      | provided container.  |
+----------------------+-------+----------------------+----------------------+

```

### CODE REVIEW

File: packs/azure/actions/lib/formatters.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+__all__ = [
+    'to_node_dict'
+]
+
+
+def to_node_dict(node):
```
Comment (Kami) [inline packs/azure/actions/lib/formatters.py]: We need to come up with a good pattern / solution for handling common code which is needed in multiple packs which doesn't result in us copy and pasting code all over the place.

Currently, the same code is needed in libcloud, softlayer and azure pack.

I was thinking about creating a new Python package - `st2contrib-libcloud-formatters` or something along those lines, but I'm not too happy with this approach.

Feedback and suggestions are welcome.

**Comment by Kami** [comment]
Also added some storage actions and updated the description.

File: packs/azure/actions/lib/formatters.py (hunk 1 above)
Comment (manasdk) [inline packs/azure/actions/lib/formatters.py]: Create an egg and install taht egg in all virtualenvs?

Comment (lakshmi-kannan) [inline packs/azure/actions/lib/formatters.py]: We can create /opt/stackstorm/libs/ and throw these common code in there. When we create virtualenvs for actions or sensors, we can create symlinks or just add this path to PYTHONPATH.

Comment (Kami) [inline packs/azure/actions/lib/formatters.py]: That's a possibility, but it doesn't (and probably shouldn't) be available to all the packs. That's why I was thinking about going with a package which is explicitly added to `requirements.txt` inside the packs which need it.

Comment (lakshmi-kannan) [inline packs/azure/actions/lib/formatters.py]: Can pip install dependencies from a file location? If so, I like the requirements approach.

**Pull request merged**